### PR TITLE
Feature use filesystemutils

### PR DESCRIPTION
**Pull request by ximex**
use the new FileSystemUtils class and some code improvements

### CODE REVIEW

**Comment by ximex** [comment]
@Fasse please review and merge if all is ok

**Comment by Fasse** [comment]
@ximex before I start testing this, have you manually test your changes. Because this is something that could break our file handling ...

**Comment by Fasse** [comment]
I will check this after commit of menu so I could check compatibility also against those changes.

**Comment by ximex** [comment]
@Fasse now you could test. Fixed some small bugs

**Comment by Fasse** [comment]
There is a problem with the htaccess file. If the file doesn't exists, it won't be created with the new class. So the folder isn't protected.

**Comment by ximex** [comment]
On my server the creation of `.htaccess` works

**Comment by Fasse** [comment]
I could reproduce it with my Mac and my Linux NAS.

To reproduce:
I have a running Admidio installation folder adm_my_files and a htaccess file.
Then I remove the htaccess file.
Now go to downloads and create new folder 
-> no htaccess file is created

**Comment by ximex** [comment]
only on update now. Look here: https://github.com/Admidio/admidio/blob/feature-use-filesystemutils/adm_program/installation/update_functions.php#L237
I could also add this code into common.php than it would be checked on any request. but i think on update it should be enough.

**Comment by Fasse** [comment]
Hmm, of you move your Installation to another webspace it should also be added. Check this at every request would be too mich, but maybe if we still handle with adm_my_files folder.

**Comment by ximex** [comment]
Or show it in the system-settings and add a button to create on if there is no `.htaccess`

**Comment by Fasse** [comment]
Ok, also possible. But Check at update should still be there.

**Comment by ximex** [comment]
Ok than i will implement this

**Comment by ximex** [comment]
@Fasse done

File: adm_program/installation/update.php [hunk 1]
```diff
@@ -102,11 +102,27 @@
...
 
+if (!is_executable(ADMIDIO_PATH . FOLDER_DATA) || !is_writable(ADMIDIO_PATH . FOLDER_DATA))
+{
+    try
+    {
+        FileSystemUtils::chmodDirectory(ADMIDIO_PATH . FOLDER_DATA, 0777);
+    }
+    catch (\RuntimeException $exception)
+    {
+        // TODO
```
Comment (Fasse) [inline adm_program/installation/update.php]: We should leave a message that the user must set the rights to **adm_my_files**.

File: adm_program/languages/en.xml [hunk 2]
```diff
@@ -1135,6 +1135,8 @@
   <string name="SYS_DELETED_USER">Deleted user</string>
   <string name="SYS_DESCRIPTION">Description</string>
   <string name="SYS_DESIGNATION">Name</string>
+  <string name="SYS_DIRECTORY_PROTECTION">Directory protection</string>
```
Comment (Fasse) [inline adm_program/languages/en.xml]: Just for info: you must only change the de.xml . Thats the only one that will be uploaded to transifex. All other files will than be downloaded from transifex.

**Comment by ximex** [comment]
@Fasse couldn't reproduce your problem. Please give more information.

**Comment by Fasse** [comment]
you can merge.

**Pull request merged**